src/components/Sidebar/Sidebar.jsx

```diff
@@ -1,15 +1,29 @@
-import React ,{useReducer} from 'react'
 import { useProduct } from '../../context/filterContext';
+import { Link } from 'react-router-dom'
 export default function Sidebar() {
 const { state, dispatch } = useProduct();
 const {sortBy,showInventoryAll,showFastDelivery,showCategory,showRating}=state
+
+const categoryHandler=(e)=>{
+const check = e.target.checked;
+const category = e.target.value;
+if (check) {
+return { type: "ADD_CATEGORY", payload: category };
+} else {
+return { type: "DELETE_CATEGORY", payload: category };
+}
+};
+
+const CheckedHandler=(category)=>{
+    return showCategory.some((item)=>item===category)
+}
 return (
 <div className="ec-side-navbar">
     <div className="ec-product-filter">
-
         <div className="ec-first-element-side-navbar">
             <div className="h3">Filters</div>
-            <a className="h5 italic-text" href="">Clear!</a>
+            <Link to='/ProductPage' onClick={()=> {
+            dispatch({ type: "FILTER_CLEAR"})}}>Clear</Link>
```

Comment:
suggestion: try to create a separate function for handling events like onchange and onclick